services/graphql-server/src/graphql/resolvers/platform/content.js

```diff
@@ -310,6 +310,14 @@ module.exports = {
    *
    */
   ContentUserRegistration: {
+    isCurrentlyRequired: async (root) => {
+      if (!root.isRequired) return false;
+      if (!root.startDate && !root.endDate) return root.isRequired;
+      const now = new Date();
+      if (root.startDate && root.startDate <= now) return false;
+      if (root.endDate && root.endDate >= now) return false;
+      return root.isRequired;
```

Comment:
I think your logic is reversed here; you want to return true if there is a start date and it's in the past. You'd need to check the end date to be sure in that condition though. Something like this?
```suggestion
      if (root.startDate && root.startDate <= now) {
        if (root.endDate && root.endDate <= now) return false;
      }
      return true;
```

Reply:
@solocommand I did have the checks wrong, but I think they still are independent conditions.  I changed it to 

```
if (root.startDate && root.startDate >= now) return false;
if (root.endDate && root.endDate <= now) return false;
```